croncat/src/streams/tasks.rs

```diff
@@ -26,14 +28,20 @@ pub async fn tasks_loop(
     let block_consumer_stream = tokio::task::spawn(async move {
         while let Ok(_block) = block_stream_rx.recv().await {
             let locked_status = block_status.lock().await;
-            if *locked_status == AgentStatus::Active {
+            let is_active = *locked_status == AgentStatus::Active;
+            // unlocking it ASAP
+            std::mem::drop(locked_status);
```

Comment:
Can you explain what's happening with this `drop`? I'm a little confused

Reply:
actually cc'ing @SeedyROM who might find this interesting

Reply:
@mikedotexe 

The drop is to free the lock before the end of the function scope. Locks drop when out of scope but sometimes you need to release a lock before that. Also `std::mem` isn't the only way to call drop. You could do `.drop()` as well.